lib/util/import-target.js

```diff
@@ -125,35 +98,196 @@ module.exports = class ImportTarget {
         this.name = name
 
         /**
-         * What type of module is this
-         * @type {'unknown'|'relative'|'absolute'|'node'|'npm'|'http'|void}
+         * The import target options.
+         * @type {Options}
          */
-        this.moduleType = "unknown"
-
-        if (name.startsWith("./") || name.startsWith(".\\")) {
-            this.moduleType = "relative"
-        } else if (name.startsWith("/") || name.startsWith("\\")) {
-            this.moduleType = "absolute"
-        } else if (isBuiltin(name)) {
-            this.moduleType = "node"
-        } else if (isNodeModule(name, options)) {
-            this.moduleType = "npm"
-        } else if (name.startsWith("http://") || name.startsWith("https://")) {
-            this.moduleType = "http"
-        }
+        this.options = options
 
         /**
-         * The full path of this import target.
-         * If the target is a module and it does not exist then this is `null`.
-         * @type {string|null}
+         * What type of module are we looking for?
+         * @type {ModuleType}
          */
-        this.filePath = getFilePath(isModule, name, options, moduleType)
+        this.moduleType = this.getModuleType()
+
+        /**
+         * What import style are we using
+         * @type {ModuleStyle}
+         */
+        this.moduleStyle = this.getModuleStyle(moduleType)
 
         /**
          * The module name of this import target.
          * If the target is a relative path then this is `null`.
-         * @type {string|null}
+         * @type {string | null}
+         */
+        this.moduleName = this.getModuleName()
+
+        /**
+         * The full path of this import target.
+         * If the target is a module and it does not exist then this is `null`.
+         * @type {string | null}
          */
-        this.moduleName = isModule ? getModuleName(name) : null
+        this.filePath = this.getFilePath()
+    }
+
+    /**
+     * What type of module is this
+     * @returns {ModuleType}
+     */
+    getModuleType() {
+        if (/^\.{1,2}([\\/]|$)/.test(this.name)) {
+            return "relative"
+        }
+
+        if (/^[\\/]/.test(this.name)) {
+            return "absolute"
+        }
+
+        if (isBuiltin(this.name)) {
+            return "node"
+        }
+
+        if (/^(@[\w~-][\w.~-]*\/)?[\w~-][\w.~-]*/.test(this.name)) {
+            return "npm"
+        }
```

Comment:
I wonder what to do with these when people use the `@organisation/package` in path aliases.